pyfar/classes/coordinates.py

```diff
@@ -1853,11 +1853,11 @@ def find_within(
                 points, distance + atol, return_sorted=return_sorted)
         if distance_measure in ['spherical_radians', 'spherical_meter']:
             # determine validate radius
-            radius = self.radius
+            radius = np.concatenate((self.radius, find.radius))
```

Comment:
There could be cases where this fails. Using this should be safe

```suggestion
            radius = np.concatenate((self.radius.flatten(), find.radius.flatten()))
```

Reply:
Ive also changed it in find_nearest